pythonforandroid/toolchain.py

```diff
@@ -578,13 +578,30 @@ def clean_recipe_build(self, args):
 
     def clean_download_cache(self, args):
         '''
-        Deletes any downloaded recipe packages.
+        Deletes a download cache for recipes stated as arguments. If no
+        argument is passed, it'll delete *all* downloaded cache. ::
+
+            p4a clean_download_cache kivy,pyjnius
 
         This does *not* delete the build caches or final distributions.
         '''
         ctx = self.ctx
-        if exists(ctx.packages_path):
-            shutil.rmtree(ctx.packages_path)
+        msg = "Download cache removed!"
+
+        if args.unknown_args:
+            for package in args.unknown_args[0].split(','):
```

Comment:
Could you replace this with a `parser_clean_download_cache.add_argument('recipes')` when the parser is defined, instead? That will mean that the user gets a helpful argparse error message if they don't explicitly pass the intended argument. You could also just allow any number of arguments (I think for argparse something like `nargs='*'`) instead of requiring the comma separation.